java-squid/src/main/java/org/sonar/java/se/checks/UnclosedResourcesCheck.java

```diff
@@ -141,12 +118,14 @@ public PostStatementVisitor(CheckerContext context) {
     public void visitNewClass(NewClassTree syntaxNode) {
       if (isOpeningResource(syntaxNode)) {
         final SymbolicValue instanceValue = programState.peekValue();
-        programState = programState.addConstraint(instanceValue, NullConstraint.OPENED);
+        if (instanceValue.wrappedValue() == instanceValue) {
```

Comment:
Here, it is important to check for the same object and not equality. How to overwrite this?